docs/utilities.rst

```diff
@@ -1,25 +1,168 @@
+
+.. Note ::
+
+    You can also override the default logging level by setting the environment variable as ``TI_LOG_LEVEL=warn``.
+
+.. function:: ti.info(info)
+
+    :parameter info: (string) logging info
+
+    Print the input string in green color to stdout **only in Taichi-scope**, when the logging level is lower or equal to ``ti.INFO``. For example:
+
+.. code-block:: python
+
+    import taichi as ti
+    ti.init(arch=ti.cpu)
+
+    ti.set_logging_level(ti.INFO)
+    var = ti.var(ti.f32, shape=1)
+
+
+    @ti.kernel
+    def compute():
+        var[0] = 1.0
+        ti.info(f"set var[0] = 1.0")
+
+
+    compute()
+
+.. function:: ti.warn(info)
+
+    :parameter info: (string) logging info
+
+    Print the input string in yellow color to stdout **only in Taichi-scope**, when the logging level is lower or equal to ``ti.WARN``.
+
+.. function:: ti.debug(info)
+
+    :parameter info: (string) logging info
+
+    Print the input string in blue color to stdout **only in Taichi-scope**, when the logging level is lower or equal to ``ti.DEBUG``.
+
+.. function:: ti.trace(info)
+
+    :parameter info: (string) logging info
+
+    Print the input string in green color to stdout **only in Taichi-scope**, only when the logging level is equal to ``ti.TRACE``.
+
+.. function:: ti.error(info)
+
+    :parameter info: (string) logging info
+
+    This function prints the input string in red color to stdout in any logging level and **only in Taichi-scope**.
+
+.. warning::
+
+    Note that ``ti.error`` will crash your program and throws an exception ``RuntimeError``.
+
+Here is an example:
+
+.. code-block:: python
+
+    import taichi as ti
+
+    ti.init()
+    ti.set_logging_level(ti.INFO)
+
+    try:
+        ti.error("It's a big error! Now exit...")
+    except RuntimeError as err:
+        print(err)
+
+
+Profiler
+--------
+
+Taichi profiler can help you count the time cost of your program. There are two profilers in Taichi available at this moment: ``ScopedProfiler`` and ``ProfilerBase``.
+
+ScopedProfiler
+##############
+
+1. ``ScopedProfiler`` evaluates the time spent for each step of the **host tasks** during execution hierarchically.
+
+2. Moreover, this profiler is turned on automatically and no extra work is needed except writing ``ti.print_profile_info()`` at the end of your Taichi program. For example:
+
+.. code-block:: python
+
+    import taichi as ti
+
+    ti.init(arch=ti.cpu)
+    var = ti.var(ti.f32, shape=1)
+
+
+    @ti.kernel
+    def compute():
+        var[0] = 1.0
+        print(f" set var[0] = {var[0]}")
+
+
+    compute()
+    ti.print_profile_info()
+
+
+``ti.print_profile_info()`` will output statistics in a hierarchical format, with different colors for different levels.
+
+.. Note::
+
+    ``ScopedProfiler`` is a c++ class inside the core of Taichi, so it is invisible in Python-scope.We mention its name here is just to make a distinction with another profiler.
+
+ProfilerBase
```

Comment:
This should be called `KernelProfiler`.

Reply:
Oh! The old `ProfilerBase` has been renamed to `KernelProfilerBase`, I didn't manage to notice it...